src/lib/autoinstall/widgets/storage/add_children_button.rb

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "cwm/common_widgets"
+
+module Y2Autoinstallation
+  module Widgets
+    module Storage
+      # This class provides a button to add 'partition' sections
+      #
+      # In an AutoYaST profile, a 'partition' section is used to define a
+      # partition, a logical volume, a RAID member, etc.
+      class AddChildrenButton < CWM::PushButton
+        extend Yast::I18n
+
+        # Constructor
+        #
+        # @param controller [Y2Autoinstallation::StorageController] UI controller
+        # @param section [Y2Storage::AutoinstProfile::DriveSection] Drive section of the profile
+        def initialize(controller, section)
+          textdomain "autoinst"
+          @controller = controller
+          @section = section
+        end
+
+        TYPE_LABELS = {
+          CT_DISK: N_("Partition")
+        }.freeze
+
+        def label
+          type_label = _(TYPE_LABELS[section.type])
+          format(_("Add %{type_label}"), type_label: type_label)
```

Comment:
To be honest I am here lost why that type label handling is needed? If it is for future extendibility how it will work with other types that can be on whole disk like LVM or BTRFS?

Reply:
Yes, it should be extended in the near future. In AutoYaST, a `drive` section can be used to define a disk (with its partitions), an LVM VG (where each partition is a logical volume), a Btrfs multi-device (which uses a partition section to define its subvolumes), a RAID (which has its own partitions) or even a Bcache device.